### PR TITLE
feat: support tcp gateways

### PR DESCRIPTION
**Pull request by randalljohnson**

### CODE REVIEW

File: internal/gateway/gateway_instance.go [hunk 1]
```diff
@@ -603,31 +588,89 @@ func confirmGatewayPortExposed(
...
+func ensureGlooEdgePortExists(protocol string, port int, tlsEnabled bool, ports []interface{}, log *logr.Logger) ([]interface{}, error) {
+
+	// check existing gateways for requested ports
+	for _, portSpec := range ports {
+		spec := portSpec.(map[string]interface{})
+		var portCurrent int64
+		portCurrent, portNumberFound, err := unstructured.NestedInt64(spec, "port")
+		if err != nil {
+			portCurrentFloat, _, errFloat := unstructured.NestedFloat64(spec, "port")
+			portCurrent = int64(portCurrentFloat)
```
Comment (randalljohnson) [inline internal/gateway/gateway_instance.go:616]: The port number is either stored as an `int64` or a `flaot64` depending on how the `GatewayDefinition` is created. E2E tests result in one outcome, while creating via a local config file results in another.

File: internal/kubernetes-runtime/mapping/location.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ func getRegionMap() *[]RegionMap {
 	return &[]RegionMap{
 		{
 			Location: "Local",
+			AwsRegion: "us-east-1",
```
Comment (randalljohnson) [inline internal/kubernetes-runtime/mapping/location.go:53]: Isn't necessary, but is helpful for testing.

File: test/e2e/e2e_workload_test.go [hunk 3]
```diff
@@ -62,23 +63,30 @@ func TestWorkloadE2E(t *testing.T) {
...
+			resp, err = http.Get(fmt.Sprintf("http://localhost:%d/version", 443))
+			// if we get an error, check HTTP
+			if err != nil {
+				resp, err = http.Get(fmt.Sprintf("http://localhost:%d/version", 80))
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		}); err != nil {
```
Comment (randalljohnson) [inline test/e2e/e2e_workload_test.go:89]: Wait until API server to start in the event that it's still performing database migrations, building etc.

File: pkg/kube/v0/namespace.go [hunk 4]
```diff
@@ -131,7 +131,11 @@ func updateNamespace(jsonDef []byte, namespace string) ([]byte, error) {
 
 	// set the namespace field in the metadata
 	if metadata, ok := mapDef["metadata"].(map[string]interface{}); ok {
-		metadata["namespace"] = namespace
+		if mapDef["kind"] == "Gateway" {
+			metadata["namespace"] = util.GatewaySystemNamespace
+		} else {
+			metadata["namespace"] = namespace
+		}
```
Comment (randalljohnson) [inline pkg/kube/v0/namespace.go:138]: We'll need a smarter way of handling this in the future

File: internal/gateway/gateway_instance.go (hunk 1 above)
Comment (randalljohnson) [reply internal/gateway/gateway_instance.go:616]: Add util.NestedIntOrFloat()

File: internal/gateway/gateway_instance.go [hunk 5]
```diff
@@ -659,84 +702,256 @@ func configureVirtualService(
...
+			}
+			if len(routes) == 0 {
+				return nil, fmt.Errorf("no routes found")
+			}
+
+			// set upstream port to 80 if tls is enabled
+			// otherwise, use the port provided
+			upstreamPort := *httpPort.Port
+			if *httpPort.TLSEnabled {
+				upstreamPort = 80
```
Comment (randalljohnson) [inline internal/gateway/gateway_instance.go:794]: mv upstreamPort containerPort

File: internal/gateway/manifests.go [hunk 6]
```diff
@@ -39,13 +43,13 @@ func createGlooEdge() (string, error) {
 				"name": "glooedge",
 			},
 			"spec": map[string]interface{}{
-				"namespace": "nukleros-gateway-system",
+				"namespace": util.GatewaySystemNamespace,
```
Comment (randalljohnson) [inline internal/gateway/manifests.go:46]: Try moving this constant to internal/gateway

File: pkg/threeport-installer/v0/support_services.go [hunk 7]
```diff
@@ -10,11 +10,13 @@ import (
...
 	v0 "github.com/threeport/threeport/pkg/api/v0"
 	kube "github.com/threeport/threeport/pkg/kube/v0"
+	util "github.com/threeport/threeport/pkg/util/v0"
 )
 
 const (
-	SupportServicesNamespace     = "support-services-system"
-	SupportServicesOperatorImage = "ghcr.io/nukleros/support-services-operator:v0.4.2"
+	SupportServicesNamespace = "support-services-system"
+	// SupportServicesOperatorImage = "ghcr.io/nukleros/support-services-operator:v0.4.2"
```
Comment (randalljohnson) [inline pkg/threeport-installer/v0/support_services.go:18]: update after support-services-operator PR is merged

**Review by lander2k2 — APPROVED** [review]

**Review by HarisChoudhary — APPROVED** [review]

**Pull request merged**